### DISCUSSION TITLE
Deterministically fetch invoices for a BOLT 12 offer

### DISCUSSION THREAD

**Discussion by ntheile** [ACINQ/phoenixd#134] [Q&A]
In core lightning I can fetch an invoice from a BOLT 12 offer like this `lightning-cli fetchinvoice lno**` and get back an invoice `lni***` that I can decode to get the payment_hash. How can I do this in phoenixd cli or api? I want to deterministically get the next 10 payment hashes for the invoices derived from the offer.

**Comment by t-bast** [comment]
As discussed on twitter, can you provide more details about why you need those and cannot do it differently? Bolt 12 invoices aren't meant to be exposed to end users and are meant to be paid immediately after generating them.

**Reply by ntheile** [reply]
My use-case is similar to this discussion https://github.com/lndk-org/lndk/issues/190. I think Strike is trying to do something similar as discussed by @mrfelton

>I've seen quite a few discussions about this, however. The spec authors argue that there is no need for the bech32 encoding for Invoices and no need for a human readable prefix because Invoices are not intended to be passed around between people.
>
>In our own use case, invoices are passed around, although not between humans. They are stored in databases and passed between systems. hex encoding does work fine for that.


@a-mpch also had a good discussion point here:
> when I started writing https://github.com/lndk-org/lndk/pull/129 the original decision was that bech32 is only used when presented to humans and spec uses TLV stream encoding for machines.
> 
> Here we are in a middle ground(?) as it is mostly expected that invoices are used by machines, but you could still use it as human using the CLI, even if is not the use case.


I will try to write up more details about my use-case in another post.
